Installation issue: libtirpc%nvhpc fails due to -pipe flag in Makefile

Attempting to build libtirpc using nvhpc compilers generates an error due to an unrecognized '-pipe' flag being passed to the
compiler.  This appears to be due to the -pipe flag being hardcoded in some Makefiles.

### Steps to reproduce the issue

<!-- Fill in the exact spec you are trying to build and the relevant part of the error message -->
```console
$ spack install libtircp@1.2.6 %nvhpc
...
==> libtirpc: Executing phase: 'build'
==> [2021-07-09-09:59:33.518276] 'make'
make  all-recursive
make[1]: Entering directory '/tmp/devspackswinst/spack-stage/spack-stage-libtirpc-1.2.6-2zmucfzyo6ukrt7qwcn4d6zlkevhibqp/spack-src'
Making all in src
make[2]: Entering directory '/tmp/devspackswinst/spack-stage/spack-stage-libtirpc-1.2.6-2zmucfzyo6ukrt7qwcn4d6zlkevhibqp/spack-src/src'
  CC       libtirpc_la-auth_none.lo
nvc-Error-Unknown switch: -pipe
make[2]: *** [Makefile:660: libtirpc_la-auth_none.lo] Error 1
```

### Information on your system
```console
> spack debug report
* **Spack:** 0.16.1-2429-f5e6c32495
* **Python:** 3.6.8
* **Platform:** linux-rhel8-x86_64
* **Concretizer:** original
```

I am using an manually installed (external to spack) nvhpc@21.5, but I expect the issue should be present with any version
of nvhpc.

### Additional information

No maintainers for libtirpc
@samcmill is the maintainer for nvhpc.

* [spack-build-env.txt](https://github.com/spack/spack/files/6792509/spack-build-env.txt)
* [spack-build-out.txt](https://github.com/spack/spack/files/6792510/spack-build-out.txt)

### General information

<!-- These boxes can be checked by replacing [ ] with [x] or by clicking them after submitting the issue. -->
- [x ] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [x ] I have run `spack maintainers <name-of-the-package>` and @mentioned any maintainers
- [x ] I have uploaded the build log and environment files
- [x ] I have searched the issues of this repo and believe this is not a duplicate

--- a/var/spack/repos/builtin/packages/libtirpc/libtirpc-remove-pipe-flag-for-nvhpc.patch
+++ b/var/spack/repos/builtin/packages/libtirpc/libtirpc-remove-pipe-flag-for-nvhpc.patch
@@ -1,0 +1,24 @@
+diff -Naur spack-src/src/Makefile.am spack-src.patch/src/Makefile.am
+--- spack-src/src/Makefile.am	2020-04-07 09:24:34.000000000 -0400
++++ spack-src.patch/src/Makefile.am	2021-07-09 10:52:47.702886134 -0400
+@@ -8,7 +8,7 @@
+ noinst_HEADERS = rpc_com.h debug.h
+ 
+ AM_CPPFLAGS = -I$(top_srcdir)/tirpc -include config.h -DPORTMAP -DINET6 \
+-		-D_GNU_SOURCE -Wall -pipe
++		-D_GNU_SOURCE -Wall
+ 
+ lib_LTLIBRARIES = libtirpc.la
+ 
+diff -Naur spack-src/src/Makefile.in spack-src.patch/src/Makefile.in
+--- spack-src/src/Makefile.in	2020-04-07 09:26:28.000000000 -0400
++++ spack-src.patch/src/Makefile.in	2021-07-09 10:52:55.102361747 -0400
+@@ -455,7 +455,7 @@
+ top_srcdir = @top_srcdir@
+ noinst_HEADERS = rpc_com.h debug.h
+ AM_CPPFLAGS = -I$(top_srcdir)/tirpc -include config.h -DPORTMAP -DINET6 \
+-		-D_GNU_SOURCE -Wall -pipe
++		-D_GNU_SOURCE -Wall
+ 
+ lib_LTLIBRARIES = libtirpc.la
+ libtirpc_la_LDFLAGS = @LDFLAG_NOUNDEFINED@ -no-undefined -lpthread \

--- a/var/spack/repos/builtin/packages/libtirpc/package.py
+++ b/var/spack/repos/builtin/packages/libtirpc/package.py
@@ -20,6 +20,9 @@ class Libtirpc(AutotoolsPackage):
 
     provides('rpc')
 
+    # Remove -pipe flag to compiler in Makefiles when using nvhpc
+    patch('libtirpc-remove-pipe-flag-for-nvhpc.patch', when='%nvhpc')
+
     # FIXME: build error on macOS
     # auth_none.c:81:9: error: unknown type name 'mutex_t'
     conflicts('platform=darwin', msg='Does not build on macOS')

